py-polars/polars/dataframe/frame.py

```diff
@@ -2286,7 +2283,40 @@ def to_pandas(  # noqa: D417
                 else:
                     raise ModuleNotFoundError(msg)
 
-        record_batches = self._df.to_pandas()
+        object_columns = [
+            name for (name, dtype) in zip(self.columns, self.dtypes) if dtype == Object
+        ]
+        object_columns_set = set(object_columns)
```

Comment:
This is better written as a set comprehension.

```suggestion
        object_columns = {
            name for (name, dtype) in zip(self.columns, self.dtypes) if dtype == Object
        }
```

Reply:
Note that the original code had a list and a set on purpose: the list is needed to preserve order, the set was needed to prevent quadratic algorithm when filtering to non-object columns. This can be relevant in cases where there are large numbers of columns.

In practice I refactored slightly to make the set unnecessary.